app/controllers/katello/api/v2/products_controller.rb

```diff
@@ -50,7 +50,7 @@ def index_relation
       query = query.enabled if params[:enabled]
       query = query.where(:id => @activation_key.products) if @activation_key
       query = query.where(:id => @system.products) if @system
-      query = query.where(:id => Pool.find_by_id!(params[:subscription_id]).products) if params[:subscription_id]
+      query = query.where(:id => Pool.find_by_cp_id(params[:subscription_id]).products) if params[:subscription_id]
```

Comment:
This seems confusing to me as a user, if I get back pool data (which in our world is subscriptions), and there is an ID and I pass that ID here I will not get my results. This being similar to the id/uuid issue we had to tackle with content types. The description of the parameter does not specify what the ID is necessarily so we could consider supporting both like we do for content types or swapping this to use our DB ID.